driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -2005,23 +2005,46 @@ public void refreshSchemaAndSignal(final Connection connection, final DefaultRes
             executor.submit(new Runnable() {
                 @Override
                 public void run() {
+                    boolean reachedSchemaAgreement = false;
                     try {
                         // Before refreshing the schema, wait for schema agreement so
                         // that querying a table just after having created it don't fail.
-                        if (!ControlConnection.waitForSchemaAgreement(connection, Cluster.Manager.this))
+                        reachedSchemaAgreement = ControlConnection.waitForSchemaAgreement(connection, Cluster.Manager.this);
+                        if (!reachedSchemaAgreement)
                             logger.warn("No schema agreement from live replicas after {} s. The schema may not be up to date on some nodes.", configuration.getProtocolOptions().getMaxSchemaAgreementWaitSeconds());
                         ControlConnection.refreshSchema(connection, keyspace, table, Cluster.Manager.this, false);
                     } catch (Exception e) {
                         logger.error("Error during schema refresh ({}). The schema from Cluster.getMetadata() might appear stale. Asynchronously submitting job to fix.", e.getMessage());
                         submitSchemaRefresh(keyspace, table);
                     } finally {
-                        // Always sets the result
+                        // Always sets the result, but remember if we reached schema agreement
+                        rs.getExecutionInfo().setReachedSchemaAgreement(reachedSchemaAgreement);
                         future.setResult(rs);
                     }
                 }
             });
         }
 
+        public void waitForSchemaAgreementAndSignal(final Connection connection, final DefaultResultSetFuture future, final ResultSet rs) {
```

Comment:
Both methods, `refreshSchemaAndSignal` and `waitForSchemaAgreementAndSignal`, are very similar.
- Would it be possible to refactor them to avoid having duplicated code?
- If I understand well, `waitForSchemaAgreementAndSignal` is used for DROPs and `refreshSchemaAndSignal` for CREATEs and UPDATEs - what is the motivation behind it? Why doesn't the driver refresh the schema on a DROP?